app/controllers/foreman_tasks/api/tasks_controller.rb

```diff
@@ -141,6 +141,31 @@ def bulk_cancel
         }
       end
 
+      api :POST, '/tasks/bulk_stop', N_('Stop all stoppable tasks')
+      param :search, String, :desc => N_('Stop tasks matching search string')
+      param :task_ids, Array, :desc => N_('Stop specific tasks by ID')
+      def bulk_stop
+        if params[:search].nil? && params[:task_ids].nil?
+          raise BadRequest, _('Please provide at least one of search or task_ids parameters in the request')
+        end
+
+        filtered_scope = bulk_scope
+        to_stop = filtered_scope.where.not(state: :stopped)
+        total_length = filtered_scope.count
+        to_stop_length = to_stop.length
+        to_stop.update_all(state: :stopped)
```

Comment:
`update_all` does not instantiate the objects and does not trigger callbacks. Because of this, the tasks wouldn't have their `state_updated_at` updated properly.

Reply:
Should I add ` state_updated_at: DateTime.now`?

Reply:
It would be best to let this trigger the callbacks, instead of specifying the date explicitly here. Using `update` on a collection or `update_attribute` on an instance might be the way to go

Reply:
changed to update as is uses validations as well